### ISSUE TITLE
Cant Delete All Messages on Wire Server

### ISSUE THREAD

**Issue by Mokda** [wireapp/wire#325]
I've got tons of messages that I wish to permanently delete but the only option **I can see** is to delete them one by one which will take too much time.

**Comment by marcoconti83** [comment]
Hi @Mokda ,
are you trying to delete messages on your mobile/computer, or on your own private backend?

If you want to delete from your mobile/computer, there is an option to "delete content" for a conversation, it will remove all of them.

**Comment by LapazV** [comment]
@marcoconti83 it is on desktop
my question is if delete content does the same thing as delete for all?
and does the other party have to do the same?

if delete content only deletes it from my side then I ask that you change this to delete at least the messages I've sent on the recipient side as well

and why cant I select multiple messages?

**Comment by LapazV** [comment]
@marcoconti83 my messages are on desktop
my question is if delete content does the same thing as delete for all?
and does the other party have to do the same?

if delete content only deletes it from my side then I ask that you change this to delete at least the messages I've sent on the recipient side as well

and why cant I select multiple messages?

**Comment by marcoconti83** [comment]
Hi @LapazV,
When you choose to "delete content", it only deletes it for you.

You could use timed messages/conversations, so that messages are automatically deleted (for everyone running a standard client) after a certain amount of time.

Alternatively, if you are using the Pro version, the creator of the conversation can delete the entire conversation for everyone. This only works for conversation that are part of a team (hence the Pro version).

**Comment by LapazV** [comment]
@marcoconti83 
hi,
this is exactly what I am talking about, with your comment you just repeated my concern

there is no way to select multiple messages and select 'delete for all' even for my own messages, there is even no way to 'select all', instead each message has a three dot menu, so now I have to do this 1000+ times to delete my own messages for all !

alternatively, you can add a feature called 'delete my content for all'

to sum up, I am asking either for 'multiple selection' of messages that has a submenu that includes 'delete for all'  in messages options, and / or make a 'delete my content for all' button, preferably both, at least for one on one conversations

**secondly,**   I am patiently waiting for the next release for desktop and there is no information on progress anywhere on your website and it has been months now, so can you please provide some information?

**Comment by LapazV** [comment]
@marcoconti83 
hi,
this is exactly what I am talking about, with your comment you just repeated my concern as I am talking about existing untimed messages

there is no way to select multiple messages and then choose 'delete for all' even for my own messages, there is even no way to 'select all', instead each message has a three dot menu, so now I have to do this 1000+ times to delete my own messages for all !

alternatively, you can add a feature called 'delete my content for all'

to sum up, I am asking either for 'multiple selection' of messages that has a submenu that includes 'delete for all'  in messages options, and / or make a 'delete my content for all' button, preferably both, at least for one on one conversations

**secondly,**   I am patiently waiting for the next release for desktop and there is no information on progress anywhere on your website and it has been months now, so can you please provide some information?

**Comment by marcoconti83** [comment]
Thanks for clarifying, this is indeed a feature that we don't have. I'll add the "feature request" label, however this is not in our roadmap for the moment.

Regarding desktop, if you have some issue specific to the desktop application, you can open a ticket on the [wire-desktop](https://github.com/wireapp/wire-desktop) repository. 

FYI, while binary releases for desktop are not frequent, the inner application on desktop (which is a JS application inside an Electron container) is updated frequently automatically without the need for you to install any new version of the binary. Every time there is an update of the [web application](https://github.com/wireapp/wire-webapp), the desktop application also receives the same update automatically. The last update was on [October 16th](https://github.com/wireapp/wire-webapp/releases/tag/2019-10-16-production.0).

**Comment by LapazV** [comment]
I would also like to add a separate point

'delete content' button has a vague label and I had to ask here with a ticket about what it actually does, so I suggest making the button's name more clear like 'delete content for me'

**Comment by LapazV** [comment]
@marcoconti83 

I would also like to add a separate point

'delete content' button has a vague label and I had to ask here with a ticket about what it actually does, so I suggest making the button's name more clear like 'delete content for me'

should I make a separate request for this? it is up to you

**Comment by LapazV** [comment]
@marcoconti83 
> Thanks for clarifying, this is indeed a feature that we don't have. I'll add the "feature request" label, however this is not in our roadmap for the moment.
> 
> Regarding desktop, if you have some issue specific to the desktop application, you can open a ticket on the [wire-desktop](https://github.com/wireapp/wire-desktop) repository.
> 
multiple selection is such a basic design issue and the lack of it causes a massive inconvenience to **such a great app** like Wire

it is like writing a 1000 lines of code to print 'hello world' instead of putting 1 line in a loop which is a huge waste of resources and a basic design fault

I really hope you would reconsider and not overlook this or push it back

please discuss this in the team and let me know what you guys think, I will wait because I really like this project

> FYI, while binary releases for desktop are not frequent, the inner application on desktop (which is a JS application inside an Electron container) is updated frequently automatically without the need for you to install any new version of the binary. Every time there is an update of the [web application](https://github.com/wireapp/wire-webapp), the desktop application also receives the same update automatically. The last update was on [October 16th](https://github.com/wireapp/wire-webapp/releases/tag/2019-10-16-production.0).

cool, I was suffering from a desktop bug that prevented me from using Wire altogether but has been fixed and I was waiting for the next release to start using it again

I will give the current version another try to see if it works now

**Comment by LapazV** [comment]
@marcoconti83 can you rename the ticket to make it more accurate for the feature?

somthing like: "Allow multiple selection of own messages in wire + delete for all"

**Comment by LapazV** [comment]
@marcoconti83 can you please rename the ticket to make it more accurate for this feature?

somthing like: "Allow multiple selection of own messages in wire + delete for all"

thanks for you help

**Comment by LapazV** [comment]
@marcoconti83 can you please rename the ticket to make it more accurate for this feature?

something like: "Allow multiple selection of own messages in wire + delete for all"

thanks for you help

**Comment by LapazV** [comment]
@marcoconti83 
can you explain what happens to untimed messages when both parties clear content?
does that clear them from the server?

**Comment by LapazV** [comment]
@marcoconti83 
can you explain what happens to untimed messages when both parties clear content or delete those messages using 'delete for me'?
does that delete messages from the server as well?

**Comment by marcoconti83** [comment]
Hi @LapazV ,
the message is never on the server in plaintext, only in encrypted format, and no one but the devices that are in the conversation at the moment have the key to decrypt that payload. The encrypted message is deleted from the backend only after enough time has passed, but no one without the proper key can decrypt it anyway. Deleting the message locally does not delete the encrypted message on the backend. You can find more details in our [security whitepaper](https://wire-docs.wire.com/download/Wire+Security+Whitepaper.pdf).

**Comment by aneverhopefulhelper** [comment]
I came to open an issue for this, but was glad to find one already exists as I too have been going through the massive amount of effort required in order to do this. Deleting every single message, one-by-one recently as I, through testing some time ago, was able to determine my messages weren't deleted off of the other person's device unless I did this and I don't trust the timed feature enough. It would be wonderful if we could have this feature and/or an update where y'all are with it.

Thank you,

-An ever hopeful helper.
